### ISSUE TITLE
fmt/std.h requires rtti

### ISSUE THREAD

**Issue by danakj** [fmtlib/fmt#3464]
Version: 10.0.0

I would like to use fmt/std.h to format std::string_view and such. I am compiling without RTTI.

But the std::exception specialization uses:
```cpp
    const std::type_info& ti = typeid(ex);
```
which does not compile with RTTI disabled in Clang or GCC.

```
/home/runner/work/subspace/subspace/third_party/fmt/include/fmt/std.h:290:32: error: use of typeid requires -frtti
    const std::type_info& ti = typeid(ex);
                               ^
```

Could we split this type which requires RTTI into a separate header?

**Comment by Dani-Hub** [comment]
The alternative solution would be to conditionally deactivate those parts of  fmt/std.h requiring RTTI when RTTI is deactivated. From a user-point of view this seems to me the better approach.

**Comment by vitaut** [comment]
Same as #3426. A PR to make RTTI usage condition (but not splitting std.h) would be welcome.

**Comment by danakj** [comment]
Ok thanks. 

Here’s a method to determine RTTI state, did not test it to verify yet: https://gist.github.com/brimston3/2be168bb423c83b0f469c0be56e66d31

**Comment by Dani-Hub** [comment]
> Ok thanks.
> 
> Here’s a method to determine RTTI state, did not test it to verify yet: https://gist.github.com/brimston3/2be168bb423c83b0f469c0be56e66d31

According to [boost/config/compiler/gcc.hpp#L122](https://github.com/boostorg/config/blob/a1cf5d531405e62927b0257b5cbecc66a545b508/include/boost/config/compiler/gcc.hpp#L122) for GCC you need to test for the GCC version as well to be sure about RTTI being enabled, but it is possible that fmtlib doesn't support older GCCs anyway (You should check).

Following further the boost RTTI detection for other compilers there are some further checks helpful (e.g. for intel there exists __INTEL_RTTI__) and others. I recommend to search for RTTI in the include/boost/config/ directory to collect them.

**Comment by danakj** [comment]
Works for MSVC, GCC, and Clang: https://godbolt.org/z/W5v595jo8

I can't test Intel on compiler explorer.

**Comment by danakj** [comment]
Compiler support:
- EDG compilers mentioned: https://github.com/fmtlib/fmt/blob/bd393456ed94a5d64bb6078aa3fa477ceeb621d9/include/fmt/format.h#LL143C17-L143C32
- GCC/Clang/MSVC mentioned in docs.
- Intel compiler in core.h

So yeah, I'll grab all of those from boost. Thanks @Dani-Hub

**Comment by danakj** [comment]
> but it is possible that fmtlib doesn't support older GCCs anyway

From https://github.com/fmtlib/fmt/blob/bd393456ed94a5d64bb6078aa3fa477ceeb621d9/doc/index.rst#portability
> These are available in GCC 4.8, Clang 3.4, MSVC 19.0 (2015) and more recent compiler version. For older compilers use {fmt} [version 4.x](https://github.com/fmtlib/fmt/releases/tag/4.1.0) which is maintained and only requires C++98.

That check is for gcc 4.3, but fmt 10+ requires GCC 4.8, so I will omit that.
